zebra-network/src/protocol/external/codec.rs

```diff
@@ -141,6 +145,10 @@ impl Encoder for Codec {
         header_writer.write_u32::<LittleEndian>(body.len() as u32)?;
         header_writer.write_all(&Sha256dChecksum::from(&body[..]).0)?;
 
+        if body.len() >= self.builder.max_len {
```

Comment:
We might want to move this to happen immediately after line 108, cc @ZcashFoundation/zebra-team

Reply:
Agree and moved at https://github.com/ZcashFoundation/zebra/pull/645/commits/d3e7fd3f8fbecb9ab1dc4ef8c462529eb375ff30